Readme.md
=================
 Büchertransport
=================

Anzeige des Büchertransportdienstes mit seinen Leihverkehrsregionen im Frontend.

--------------
 Beschreibung
-------------- 
Die Extension Büchertransport ist dafür vorgesehen alle Funktionalitäten und Informationen von der alten statischen Büchertransportdienst-Website in TYPO3 zu integrieren. Darunter fallen die Einzelinformationen zu den jeweiligen Leihverkehrsregionen (Beschreibungstexte, Transportrouten, Hauptumschlagspunkte und angeschlossene sowie erreichbare Bibliotheken) mit einer Kartendarstellung der Transportwege.

Die Daten für den Büchertransportdienst werden dazu in drei Klassen unterteilt: Province (Leihverkehrsregion), City (Stadt) und Library (Bibliothek), die über das TYPO3-Backend gepflegt werden können. Die Daten sind in einem Systemordner 'Buechertransport' hinterlegt. 


Changelog
2013-04-21 - 0.3 - Dominic Simm <simm@sub.uni-goettingen.de>
	* Rewrite for TYPO3 6.0 because of compatibility problems with repositories
	* Removed t3jquery dependency

2012-12-19 - 0.2.6 - Dominic Simm <simm@sub.uni-goettingen.de>
	* Added Provinces Bremen (Niedersachsen) and Rheinland (NRW)
	* Changes in manual.rst

2012-12-18 - 0.2.5 - Dominic Simm <simm@sub.uni-goettingen.de>
	* Added nkwlib-include to provinceController [->GeoCodes]
	* Removed persistence.storagePid from Typoscript

2012-12-14 - 0.2.4 - Dominic Simm <simm@sub.uni-goettingen.de>
	* Added fields 'geocode', lat' and 'lng' to province-model

2012-12-14 - 0.2.3 - Dominic Simm <simm@sub.uni-goettingen.de>
	* Added GeocodeCommandController-action for scheduled geocode-addition for provinces

2012-12-14 - 0.2.2 - Dominic Simm <simm@sub.uni-goettingen.de>
	* Added getter-functions to province-model for local and global reachable distcentres
	* Some style changes

2012-12-14 - 0.2.1 - Dominic Simm <simm@sub.uni-goettingen.de>
	* Added fields 'lat','lng' to city-model

2012-12-13 - 0.2.0 - Dominic Simm <simm@sub.uni-goettingen.de>
	* Added GeocodeCommandController for scheduled geocode-addition
	* Added field 'geocode' to city-model
	* Changed reachable cities relation in Province-model to m:n
	* Changed GMaps-template for automated direction inclusion
	* [Feature] Added FusionTable Support for Province Borders
	* Added Google Maps support for presentation of 'Transportwege'

2012-12-04 - 0.1.9 - Dominic Simm <simm@sub.uni-goettingen.de>
	* added jQuery-TableSorter
	
2012-11-30 - 0.1.8 - Dominic Simm <simm@sub.uni-goettingen.de>
	* changed importer to import only all cities and libraries of a region added DNB-'Sigel' link
	* implemented sidebar hook; corrected importer to add also reachable cities; changed persistence model from m:n to 1:n

2012-11-28 - 0.1.7 - Dominic Simm <simm@sub.uni-goettingen.de>
	* added province dependence-files
	* added cHash to navigation links

2012-11-26 - 0.1.6 - Dominic Simm <simm@sub.uni-goettingen.de>
	* repaired navigation links [added correct pid]
	* [Bug] caching problem

2012-11-18 - 0.1.5 - Dominic Simm <simm@sub.uni-goettingen.de>
	* added Typoscript to include navigation and breadcrumb
	* modified userFunction [navigation helper class]

2012-11-14 - 0.1.4 - Dominic Simm <simm@sub.uni-goettingen.de>
	* added helper class to produce nav-menu
	* added distCentre to importer

2012-11-14 - 0.1.3 - Dominic Simm <simm@sub.uni-goettingen.de>
	* added further translations
	* changed show / list view
	* added build files

2012-11-13 - 0.1.2 - Dominic Simm <simm@sub.uni-goettingen.de>
	* added backend translations
	* added backend icons, header image
	* improved import performance (removed unnecessary repository-calls)
	* added documentation/manual under doc
	* restructed models (added mm-relations, but needs further changes)

2012-11-13 - 0.1.1 - Dominic Simm <simm@sub.uni-goettingen.de>
	* Bugfix: Removed lazy loading of province cities (Scheduler: Repository-Update-Problem)

2012-11-12 - 0.1.0 - Dominic Simm <simm@sub.uni-goettingen.de>
	* initial import


Configuration/TCA/Province.php
<?php
if (!defined ('TYPO3_MODE')) {
	die ('Access denied.');
}

$TCA['tx_buechertransport_domain_model_province'] = array(
	'ctrl' => $TCA['tx_buechertransport_domain_model_province']['ctrl'],
	'interface' => array(
		'showRecordFieldList' => 'sys_language_uid, l10n_parent, l10n_diffsource, hidden, name, description, geocode, lat, lng, cities, reachables',
	),
	'types' => array(
		'1' => array('showitem' => 'sys_language_uid;;;;1-1-1, l10n_parent, l10n_diffsource, hidden;;1, name, description, geocode, lat, lng, cities, reachables,--div--;LLL:EXT:cms/locallang_ttc.xlf:tabs.access,starttime, endtime'),
	),
	'palettes' => array(
		'1' => array('showitem' => ''),
	),
	'columns' => array(
		'sys_language_uid' => array(
			'exclude' => 1,
			'label' => 'LLL:EXT:lang/locallang_general.xlf:LGL.language',
			'config' => array(
				'type' => 'select',
				'foreign_table' => 'sys_language',
				'foreign_table_where' => 'ORDER BY sys_language.title',
				'items' => array(
					array('LLL:EXT:lang/locallang_general.xlf:LGL.allLanguages', -1),
					array('LLL:EXT:lang/locallang_general.xlf:LGL.default_value', 0)
				),
			),
		),
		'l10n_parent' => array(
			'displayCond' => 'FIELD:sys_language_uid:>:0',
			'exclude' => 1,
			'label' => 'LLL:EXT:lang/locallang_general.xlf:LGL.l18n_parent',
			'config' => array(
				'type' => 'select',
				'items' => array(
					array('', 0),
				),
				'foreign_table' => 'tx_buechertransport_domain_model_province',
				'foreign_table_where' => 'AND tx_buechertransport_domain_model_province.pid=###CURRENT_PID### AND tx_buechertransport_domain_model_province.sys_language_uid IN (-1,0)',
			),
		),
		'l10n_diffsource' => array(
			'config' => array(
				'type' => 'passthrough',
			),
		),
		't3ver_label' => array(
			'label' => 'LLL:EXT:lang/locallang_general.xlf:LGL.versionLabel',
			'config' => array(
				'type' => 'input',
				'size' => 30,
				'max' => 255,
			)
		),
		'hidden' => array(
			'exclude' => 1,
			'label' => 'LLL:EXT:lang/locallang_general.xlf:LGL.hidden',
			'config' => array(
				'type' => 'check',
			),
		),
		'starttime' => array(
			'exclude' => 1,
			'l10n_mode' => 'mergeIfNotBlank',
			'label' => 'LLL:EXT:lang/locallang_general.xlf:LGL.starttime',
			'config' => array(
				'type' => 'input',
				'size' => 13,
				'max' => 20,
				'eval' => 'datetime',
				'checkbox' => 0,
				'default' => 0,
				'range' => array(
					'lower' => mktime(0, 0, 0, date('m'), date('d'), date('Y'))
				),
			),
		),
		'endtime' => array(
			'exclude' => 1,
			'l10n_mode' => 'mergeIfNotBlank',
			'label' => 'LLL:EXT:lang/locallang_general.xlf:LGL.endtime',
			'config' => array(
				'type' => 'input',
				'size' => 13,
				'max' => 20,
				'eval' => 'datetime',
				'checkbox' => 0,
				'default' => 0,
				'range' => array(
					'lower' => mktime(0, 0, 0, date('m'), date('d'), date('Y'))
				),
			),
		),
		'name' => array(
			'exclude' => 0,
			'label' => 'LLL:EXT:buechertransport/Resources/Private/Language/locallang_db.xlf:tx_buechertransport_domain_model_province.name',
			'config' => array(
				'type' => 'input',
				'size' => 30,
				'eval' => 'trim,required'
			),
		),
		'description' => array(
			'exclude' => 0,
			'label' => 'LLL:EXT:buechertransport/Resources/Private/Language/locallang_db.xlf:tx_buechertransport_domain_model_province.description',
			'config' => array(
				'type' => 'text',
				'cols' => 40,
				'rows' => 15,
				'eval' => 'trim,required',
				'wizards' => array(
					'RTE' => array(
						'icon' => 'wizard_rte2.gif',
						'notNewRecords'=> 1,
						'RTEonly' => 1,
						'script' => 'wizard_rte.php',
						'title' => 'LLL:EXT:cms/locallang_ttc.xlf:bodytext.W.RTE',
						'type' => 'script'
					)
				)
			),
			'defaultExtras' => 'richtext:rte_transform[flag=rte_enabled|mode=ts]',
		),
		'geocode' => array(
			'exclude' => 0,
			'label' => 'LLL:EXT:buechertransport/Resources/Private/Language/locallang_db.xlf:tx_buechertransport_domain_model_province.geocode',
			'config' => array(
				'type' => 'input',
				'size' => 30,
				'eval' => 'trim'
			),
		),
		'lat' => array(
			'exclude' => 0,
			'label' => 'LLL:EXT:buechertransport/Resources/Private/Language/locallang_db.xlf:tx_buechertransport_domain_model_province.lat',
			'config' => array(
				'type' => 'input',
				'size' => 30,
				'eval' => 'trim'
			),
		),
		'lng' => array(
			'exclude' => 0,
			'label' => 'LLL:EXT:buechertransport/Resources/Private/Language/locallang_db.xlf:tx_buechertransport_domain_model_province.lng',
			'config' => array(
				'type' => 'input',
				'size' => 30,
				'eval' => 'trim'
			),
		),
		'cities' => array(
			'exclude' => 0,
			'label' => 'LLL:EXT:buechertransport/Resources/Private/Language/locallang_db.xlf:tx_buechertransport_domain_model_province.cities',
			'config' => array(
				'type' => 'inline',
				'foreign_table' => 'tx_buechertransport_domain_model_city',
				'foreign_field' => 'province',
				'maxitems'      => 9999,
				'appearance' => array(
					'collapseAll' => 1,
					'levelLinksPosition' => 'top',
					'showSynchronizationLink' => 1,
					'showPossibleLocalizationRecords' => 1,
					'showAllLocalizationLink' => 1
				),
			),
		),
		'reachables' => array(
			'exclude' => 0,
			'label' => 'LLL:EXT:buechertransport/Resources/Private/Language/locallang_db.xlf:tx_buechertransport_domain_model_province.reachables',
			'config' => array(
				'type' => 'select',
				'foreign_table' => 'tx_buechertransport_domain_model_city',
				'MM' => 'tx_buechertransport_province_reachables_city_mm',
				'size' => 10,
				'autoSizeMax' => 30,
				'maxitems' => 9999,
				'multiple' => 0,
				'wizards' => array(
					'_PADDING' => 1,
					'_VERTICAL' => 1,
					'edit' => array(
						'type' => 'popup',
						'title' => 'Edit',
						'script' => 'wizard_edit.php',
						'icon' => 'edit2.gif',
						'popup_onlyOpenIfSelected' => 1,
						'JSopenParams' => 'height=350,width=580,status=0,menubar=0,scrollbars=1',
						),
					'add' => Array(
						'type' => 'script',
						'title' => 'Create new',
						'icon' => 'add.gif',
						'params' => array(
							'table' => 'tx_buechertransport_domain_model_city',
							'pid' => '###CURRENT_PID###',
							'setValue' => 'prepend'
							),
						'script' => 'wizard_add.php',
					),
				),
			),
		),
	),
);

?>